projects/dsp-ui/src/lib/viewer/views/property-view/property-view.component.spec.ts

```diff
@@ -24,40 +31,34 @@ class TestPropertyParentComponent implements OnInit {
 
   propArray: PropertyInfoValues[] = [];
 
-  systemPropArray?: PropertyDefinition[] = [];
+  systemPropArray: SystemPropertyDefinition[] = [];
 
   ngOnInit() {
 
     MockResource.getTestthing().subscribe(response => {
-      this.parentResource = response;
-      const propsList: IHasProperty[] = this.parentResource.entityInfo.classes[this.parentResource.type].propertiesList;
-
-      for (const prop of propsList) {
-        const index = prop.propertyIndex;
-
-        if (this.parentResource.entityInfo.properties[index]) {
-          if (this.parentResource.entityInfo.properties[index] instanceof ResourcePropertyDefinition) {
-            // filter all properties by type ResourcePropertyDefinition
-            const propInfoAndValues: PropertyInfoValues = {
-              guiDef: prop,
-              propDef: this.parentResource.entityInfo.properties[index],
-              values: this.parentResource.properties[index]
-            };
-
-            this.propArray.push(propInfoAndValues);
-
-          } else if (this.parentResource.entityInfo.properties[index] instanceof SystemPropertyDefinition) {
-            // filter all properties by type SystemPropertyDefinition
-            const systemPropInfo = this.parentResource.entityInfo.properties[index];
-
-            this.systemPropArray.push(systemPropInfo);
-
-          }
-
-        } else {
-          console.error('Error detected: the property with IRI =' + index + 'is not a property of the resource');
-        }
-      }
+        this.parentResource = response;
+
+        // gather resource property information
+        this.propArray = this.parentResource.entityInfo.classes[this.parentResource.type].getResourcePropertiesList().map(
```

Comment:
Do I understand correctly that you have to mock the information that would be passed in from `ResourceViewComponent`? And you do this in `ResourceViewComponent` and not in `PropertyViewComponent` because otherwise the same thing would have to be repeated for each property?

I have a similar problem in the JS lib: the mock involves some complexity. I think your approach is totally fine for now, but could you create a DSP task so we can evaluate:
- whether to centralise this in UI lib (this would also involve some thinking about whether in specs we should use productive code or not at all)
- move this logic to JS lib to so the information would be returned with a mocked resource

Reply:
Yes, you understood it correctly. I even asked for your opinion on this when I created the mock and it tests and you agreed. I can create a DSP issue, no problem.

Reply:
[DSP-282](https://dasch.myjetbrains.com/youtrack/issue/DSP-282)

Reply:
>  I even asked for your opinion on this when I created the mock and it tests and you agreed.

ok :-)